var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -233,7 +238,8 @@ def cmake_args(self):
                 define(project_name_base + "_ENABLE_SEACAS", True),
             ]
         )
-
+        if "~shared" in self.spec:
+            options.append(self.define("Seacas_EXTRA_LINK_FLAGS", "z;dl"))
```

Comment:
This should also probably be `options.append(self.define(project_name_base + "_EXTRA_LINK_FLAGS", "z;dl"))`